data/adapters/ontologies_adapter.py

```diff
@@ -134,17 +136,42 @@ def process_edges(self, predicate):
                 nodes.add(to_node)
 
             if self.type == 'edge':
+                from_node_key = Ontology.to_key(from_node)
+                predicate_key = Ontology.to_key(predicate)
+                to_node_key = Ontology.to_key(to_node)
+
+                if predicate == Ontology.DB_XREF:
+                    if to_node.__class__ == rdflib.term.Literal:
+                        if str(to_node) == str(from_node):
+                            print('Skipping self xref for: ' + from_node_key)
+                            continue
+
+                        # only accepting IDs in the form <ontology>:<ontology_id>
+                        if len(str(to_node).split(':')) != 2:
+                            print(
+                                'Unsupported format for xref: ' + str(to_node))
+                            continue
+
+                        to_node_key = str(to_node).replace(':', '_')
+
+                        if from_node_key == to_node_key:
```

Comment:
is `if str(to_node) == str(from_node):` duplicated with `if from_node_key == to_node_key:`?

Reply:
it looks duplicated, but once you replace ':' by '_' you can still have duplicates. These are edge cases.